compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -861,31 +860,32 @@ private BLangStatementExpression createIfElseFromConfigurable(BLangSimpleVariabl
     }
 
     private List<BLangVariable> desugarGlobalVariables(BLangPackage pkgNode, BLangBlockFunctionBody initFnBody) {
-        List<BLangVariable> globalVars = pkgNode.globalVars;
         List<BLangVariable> desugaredGlobalVarList = new ArrayList<>();
         SymbolEnv initFunctionEnv =
                 SymbolEnv.createFunctionEnv(pkgNode.initFunction, pkgNode.initFunction.symbol.scope, env);
 
-        globalVars.forEach(globalVar -> {
+        for (BLangVariable globalVar : pkgNode.globalVars) {
             this.env.enclPkg.topLevelNodes.remove(globalVar);
             // This will convert complex variables to simple variables.
             switch (globalVar.getKind()) {
                 case TUPLE_VARIABLE:
+                case RECORD_VARIABLE:
+                case ERROR_VARIABLE:
                     BLangNode blockStatementNode = rewrite(globalVar, initFunctionEnv);
                     List<BLangStatement> statements = ((BLangBlockStmt) blockStatementNode).stmts;
                     for (int i = 0; i < statements.size(); i++) {
                         BLangStatement bLangStatement = statements.get(i);
                         // First statement is the virtual array created for the init expression.
-                        // Rest binding pattern array initialization will be desugared as a block hence add them
-                        // directly to the init function body.
+                        // Rest binding pattern will be desugared as a block hence add them directly to the
+                        // init function body.
                         if (bLangStatement.getKind() == NodeKind.BLOCK || i == 0) {
```

Comment:
What if instead of desugaring the rest part to a separate block, we check if the statement is a var def statement, and then check `(((BLangSimpleVariableDef) bLangStatement).var.symbol.owner & SymTag.PACKAGE) == SymTag.PACKAGE` here to do this differentiation?